### DISCUSSION TITLE
Issues when running the code in Colab

### DISCUSSION THREAD

**Discussion by Jose-Augusto-C-M** [open-mmlab/mmdetection#7841] [Q&A]
Hello, first I would like to thank you guys for all the hard work put in mmlab in general, It´s an amazing contribution.

For the error, I´m try to run the codes: '[MMDet_InstanceSeg_Tutorial.ipynb] and [MMDet_Tutorial.ipynb]' on colab, in both I´m facing an error when running this part of the code:

'
from mmdet.datasets import build_dataset
from mmdet.models import build_detector
from mmdet.apis import train_detector

# Build dataset
datasets = [build_dataset(cfg.data.train)]

# Build the detector
model = build_detector(cfg.model)

# Add an attribute for visualization convenience
model.CLASSES = datasets[0].CLASSES

# Create work_dir
mmcv.mkdir_or_exist(osp.abspath(cfg.work_dir))
train_detector(model, datasets, cfg, distributed=False, validate=True)
'

The result of the code is:
'

loading annotations into memory...
Done (t=0.01s)
creating index...
index created!
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
[<ipython-input-17-e916bd97cdb5>](https://localhost:8080/#) in <module>()
     15 # Create work_dir
     16 mmcv.mkdir_or_exist(osp.abspath(cfg.work_dir))
---> 17 train_detector(model, datasets, cfg, distributed=False, validate=True)

2 frames
[/content/mmdetection/mmdet/apis/train.py](https://localhost:8080/#) in train_detector(model, dataset, cfg, distributed, validate, timestamp, meta)
    161             find_unused_parameters=find_unused_parameters)
    162     else:
--> 163         model = build_dp(model, cfg.device, device_ids=cfg.gpu_ids)
    164 
    165     # build optimizer

[/usr/local/lib/python3.7/dist-packages/mmcv/utils/config.py](https://localhost:8080/#) in __getattr__(self, name)
    511 
    512     def __getattr__(self, name):
--> 513         return getattr(self._cfg_dict, name)
    514 
    515     def __getitem__(self, name):

[/usr/local/lib/python3.7/dist-packages/mmcv/utils/config.py](https://localhost:8080/#) in __getattr__(self, name)
     47         else:
     48             return value
---> 49         raise ex
     50 
     51 

**AttributeError: 'ConfigDict' object has no attribute 'device'**
'

This happens when using the exact code from the demo, I did not change anything. What should I do?

**Comment by 12567878** [comment]
I came up with the same trouble.
In mmdet/apis/train.py:163  change it to:
model = build_dp(model,  device_ids=cfg.gpu_ids) # delete parameter cfg.device,it will work.

API:def build_dp(model, device='cuda', dim=0, *args, **kwargs):
Passing cfg.device='cuda' may work,too.

**Comment by Jose-Augusto-C-M** [comment]
Thank you bro or sister 😅, I will come back to investigate the issues with
my code, there's some time that i don't revisit this demand.

But very fucking thank you for you help. 🖖

**Comment by ai21mtech13002** [comment]
Hi, I cam across the same issue. I was bale to resolve it be setting 
cfg.device='cuda' in my configuration script
